snap/integrity/dmverity/veritysetup.go

```diff
@@ -81,11 +84,62 @@ func getRootHashFromOutput(output []byte) (rootHash string, err error) {
 	return rootHash, nil
 }
 
+func verityVersion() (major, minor, patch int, err error) {
+	cmd := exec.Command("veritysetup", "--version")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return -1, -1, -1, osutil.OutputErr(output, err)
+	}
+
+	exp := regexp.MustCompile(`(\d+)\.(\d+)\.(\d+)`)
```

Comment:
Hmm I personally feel like it's completely unnecessary if we look at their versioning (https://gitlab.com/cryptsetup/cryptsetup/-/tree/master/docs?ref_type=heads). They use 3 componenet versioning, and maybe it's better we start out strict for this, and should it ever be an issue we can revisit this. 

I'm not sure about the regex, but the unit tests work as expected unless I'm missing something (anything not X.Y.Z returns an error)? I think this needs a bit more elaboration

Reply:
in theory it should be `(\d+)\.?(\d+)?\.?(\d+)?` if thinks like `1` or `1.2` going to be accepted, otherwise the 3 elements are mandatory - that's why I say that it is strange that the unit tests are fine with versions like that. For instance, using `(\d+)\.(\d+)\.(\d+)` in https://regex101.com/ does not match `1.2`

Reply:
As discussed this was intentional